example/main.go

```diff
@@ -12,41 +11,40 @@ import (
 
 func main() {
 	vecty.RenderBody(&body{
-		store: vstore.CreateStore(&store.State{}),
+		store: vstore.New(&store.State{}),
 	})
 }
 
 type body struct {
 	vecty.Core
-	store *vstore.Store
+	store vstore.Store
 }
 
 func (b *body) Render() *vecty.HTML {
 	return elem.Body(
-		b.store.NewComponent(NewMainComp),
+		b.store.Connect(NewMainComp()),
 	)
 }
 
 type mainComp struct {
 	vecty.Core
-	Number   int `vecty:"prop"`
-	Dispatch func(action interface{})
+	store vstore.Store
```

Comment:
One neat feature of react's component-structure is the ability to declare what data a component requires in order to be rendered correctly. This is done through `ReactProps` since JavaScript is not a statically typed language.

However, with Go, we can declare our data-dependencies in the struct Fields, which is even better since the properties are checked at compile time. Therefore, you can just look at a component `struct` and see what kind of data it requires.

Therefore, `vstore` shouldn't take away this feature form the user. So we should put back the `Number` field here and populate it in the `Connect` function. This means that `Dispatch` will need to call `Connect` on every component before it calls `go vecty.Rerender(comp)`. 


This is even better than my original solution because we won't even need to add the `vecty:"prop"` struct tag anymore.

Reply:
I'd like the user to have the option to have the store or state as a component struct field. Do I need to change when connect is called for the case of the component having the state as a struct field?